src/course-unit/data/thunk.js

```diff
@@ -27,6 +47,22 @@ export function fetchCourseUnitQuery(courseId) {
   };
 }
 
+export function fetchCourseSectionVerticalData(courseId) {
```

Comment:
[important] we can use createAsyncThunk from reduxjs/toolkit to reduce the boilerplate (remove unnecessary exports from slice, reduce code for thunk file). RTK will handle success, failure, rejected statuses by itself.

as example
```
export const fetchCourseUnit = createAsyncThunk(
  'courseUnit/fetchCourseUnit',
  async (courseId, { rejectWithValue }) => {
    try {
      return await getCourseUnitData(courseId);
    } catch (error) {
      return rejectWithValue(error);
    }
  }
);
```

and for slice 

```
const slice = createSlice({
  name: 'courseUnit',
  initialState: {},
  reducers: // replace from reducers success failed cases to extraReducers,
  extraReducers: (builder) => {
    builder
      .addCase(fetchCourseUnit.pending, (state) => {/*...*/})
      .addCase(fetchCourseUnit.fulfilled, (state, action) => {/*...*/})
      .addCase(fetchCourseUnit.rejected, (state, action) => {/*...*/});
  }
});
```